backend/penndata/views.py

```diff
@@ -70,80 +79,19 @@ def get(self, request):
-        # finds table with all information and gets the rows
-        table = soup.find(
-            "table",
-            {
-                "class": (
-                    "table table-bordered table-striped "
-                    "table-condensed table-responsive calendar-table"
-                )
-            },
-        )
-        rows = table.find_all("tr")
-        calendar = []
-        current_year = timezone.localtime().year
-        row_year = 0
-
-        # collect end dates on all events and filter based on that
-        for row in rows:
-            header = row.find_all("th")
-            if len(header) > 0:
-                row_year = header[0].get_text().split(" ")[0]
-            # skips calculation if years don't align
-            if int(row_year) != current_year:
-                continue
-            if len(header) == 0:
-                data = row.find_all("td")
-                date_info = data[1].get_text()
-                date = None
-                try:
-                    # handles case for date ex. August 31
-                    date = datetime.datetime.strptime(
-                        date_info + str(current_year) + "-04:00", "%B %d%Y%z"
-                    )
-                except ValueError:
-                    try:
-                        # handles case for date ex. August 1-3
-                        month = date_info.split(" ")[0]
-                        day = date_info.split("-")[1]
-                        date = datetime.datetime.strptime(
-                            month + day + str(current_year) + "-04:00", "%B%d%Y%z"
-                        )
-                    except (ValueError, IndexError):
-                        try:
-                            # handles case for date ex. August 1-September 31
-                            last_date = date_info.split("-")[0].split(" ")
-                            month = last_date[0]
-                            day = last_date[1]
-                            date = datetime.datetime.strptime(
-                                month + day + str(current_year) + "-04:00", "%B%d%Y%z"
-                            )
-                        except (ValueError, IndexError):
-                            pass
-
-                # TODO: add this: and date < timezone.localtime() + datetime.timedelta(days=14)
-                if date and date > timezone.localtime():
-                    calendar.append({"event": data[0].get_text(), "date": data[1].get_text()})
-                # only returns the 3 most recent events
-                if len(calendar) == 3:
-                    break
-        return calendar
+    permission_classes = [AllowAny]
+    serializer_class = CalendarEventSerializer
 
-    def get(self, request):
-        return Response(self.get_calendar())
+    def get_queryset(self):
+        return CalendarEvent.objects.filter(
+            date_obj__gte=timezone.localtime(),
+            date_obj__lte=timezone.localtime() + timedelta(days=30),
```

Comment:
Wait also if it is only called once a month, then by the end of the month there won't be many future events left in the db, so we wouldn't really have any calendar events?

Reply:
Ah yes good point. I'll change the `get_calendar` to create `CalendarEvents` that exceed 30 days as well. Then we will leave this view as is.